front-end/src/renderer/components/RunningClockDatePicker.vue

```diff
@@ -35,6 +35,22 @@ function stopInterval() {
   intervalId.value && clearInterval(intervalId.value);
 }
 
+function resetTime(date: Date) {
+  const newDate = new Date(date);
+  newDate.setHours(0, 0, 0, 0);
+  return newDate;
+}
+
+function handleUpdateValidStart(date: Date) {
+  const isNewDate = date.toDateString() !== props.modelValue.toDateString();
```

Comment:
If the date selected is today's date (whether using the 'now' button, or manually selecting today), then 'isNewDate' should be false. While the RunningClockDatePicker will see the time is past and update it to the current time, visually it is a bit jarring for the user to see they selected today, and the time is 00:00:00.